main/src/main/java/cgeo/geocaching/connector/gc/ImportBookmarkLinks.java

```diff
@@ -9,13 +9,90 @@
 import android.net.Uri;
 import android.os.Bundle;
 
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 
 import java.util.regex.Pattern;
 
+import javax.annotation.RegEx;
+
 public class ImportBookmarkLinks extends AppCompatActivity {
 
+    private final UrlToIdParser intentUrlParser = defaultBookmarkListUrlToIdParser();
+
+    public static final String BOOKMARK_LIST_API_PREFIX =
+            "https://www.geocaching.com/plan/api/gpx/list/";
+
+    interface UrlToIdParser {
```

Comment:
No need to make up an interface for this, you may use the generic functional interface `cgeo.geocaching.utils.functions.Func1` 

(NB: I just noted that apparently my IDE does not complain the usage of the java.util.function interfaces any more like it used to. Has something changed here? I will investigate, maybe we can finally get rid of cgeo.geocaching.utils.functions.... :-))

Reply:
I still get the warning:
`Class requires API level 24 (current min is 21): java.util.function.Function`

Also: I made the interface to be domain-specific. It signals a specific responsibility. I have, however, now made it extend `Func1`